test/extended/router/certgen/certgen.go

```diff
@@ -0,0 +1,119 @@
+package certgen
+
+import (
+	"bytes"
+	"crypto/ecdsa"
+	"crypto/elliptic"
+	"crypto/rand"
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"encoding/pem"
+	"fmt"
+	"math/big"
+	"net"
+	"time"
+)
+
+// MarshalPrivateKeyToDERFormat converts the key to a string
+// representation (SEC 1, ASN.1 DER form) suitable for dropping into a
+// route's TLS key stanza.
+func MarshalPrivateKeyToDERFormat(key *ecdsa.PrivateKey) (string, error) {
+	data, err := x509.MarshalECPrivateKey(key)
```

Comment:
Is there any specific reason to use EC rather than RSA keys?

Reply:
No, just code that I used first time around when I created the tests.